Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -194,10 +205,15 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian_2nd(
                     if (mag_exchange_coupling == 1){
                         // H_exchange
                         if (mag_exchange_arrx == 0._rt) amrex::Abort("The mag_exchange_arrx is 0.0 while including the exchange coupling term H_exchange for H_eff");
+
+                        LapM_old_xface(i, j, k, 0) = T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 0); // M_xface is not updated yet, i.e. M = M_old 
+                        LapM_old_xface(i, j, k, 1) = T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 1); // M_xface is not updated yet, i.e. M = M_old 
+                        LapM_old_xface(i, j, k, 2) = T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 2); // M_xface is not updated yet, i.e. M = M_old 
+
                         amrex::Real const H_exchange_coeff = 2.0 * mag_exchange_arrx / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx;
-                        Hx_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 0);
-                        Hy_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 1);
-                        Hz_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 2);
+                        Hx_eff += H_exchange_coeff * LapM_old_xface(i, j, k, 0);
+                        Hy_eff += H_exchange_coeff * LapM_old_xface(i, j, k, 1);
+                        Hz_eff += H_exchange_coeff * LapM_old_xface(i, j, k, 2);
```

Comment:
same here.

Reply:
I think in the iterations we want to use (1/2) * (LapM_old + LapM_prev) so we may want to keep around LapM_old instead of re-computing it.

Reply:
wait... let me review the algorithm this may not be necessary

Reply:
Yeah I also had to review our paper to realize we actually don't need `(1/2) * (LapM_old + LapM_prev)`. `LapM_old ` only appears in `b_temp_static`, and `LapM_prev ` only appears in `a_temp_dynamic`

Reply:
yes